### PR TITLE
Kinectograph

### PR DESCRIPTION
**Pull request by aMarcireau**

### CODE REVIEW

File: python/faery/events_stream.py [hunk 1]
```diff
@@ -439,22 +509,61 @@ def apply(
...
+        state_manager.end()
+        return result
 
     def envelope(
         self,
         decay: enums.Decay,
         tau: timestamp.Time,
     ) -> frame_stream.FiniteRegularFloat64FrameStream: ...
 
+    def to_kinectograph(
```
Comment (Jegp) [inline python/faery/events_stream.py:536]: I like that the work is done in a class - that we can then "pull into" the streaming API. That makes the code much less messy

**Review by Jegp — APPROVED** [review]
Overall, fantastic input. As always :-)

Wouldn't it be helpful to add a python test or two in the `tests` folder? Not necessarily super complex tests. Mostly tests that would demonstrate that a `to_kinectograph` works in the future, for regression reasons. The codebase is becoming sufficiently complex that I would be worried about breaking something because I loose oversight. 

Separately, it would be great to have just a few lines in a docstring for the classes and methods. I'm not sure everyone knows what a kinectograph is. But this is also related to #4

File: python/faery/task.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+from . import display, timestamp
+
+if typing.TYPE_CHECKING:
+    from .types import job_metadata  # type: ignore
+else:
+    from .extension import job_metadata
+
+
+@functools.cache
+def dirname() -> pathlib.Path:
```
[the remark this answers is not in the document]
Comment (aMarcireau) [reply python/faery/task.py]: Fixed in bc42d872b36f4ba91b4f0c765772e445642b809c.

**Comment by aMarcireau** [comment]
> Overall, fantastic input. As always :-)
> 
> Wouldn't it be helpful to add a python test or two in the `tests` folder? Not necessarily super complex tests. Mostly tests that would demonstrate that a `to_kinectograph` works in the future, for regression reasons. The codebase is becoming sufficiently complex that I would be worried about breaking something because I loose oversight.
> 
> Separately, it would be great to have just a few lines in a docstring for the classes and methods. I'm not sure everyone knows what a kinectograph is. But this is also related to #4

I agree with the need for docstrings but I am currently facing an issue. The user-facing `to_kinectograph` function is defined four times (once per stream type). Whilst they share the same "implementation" (with some dynamic binding magic), comments evaluated statically would need to be duplicated four times.

I created a bunch of examples (in the examples directory) that we could upgrade to tests (either by running pytest on the examples directory or by moving the files to the tests directory). Unlike tests, however, the results are not automatically checked for inconsistencies (partly because this is hard to do with images or videos).

**Pull request merged**